### PR TITLE
Fix Missions 

### PR DESCRIPTION
**Pull request by DoctaEnkoda**
Revert partiel Iridium-Development/IridiumSkyblock#384

Need more test

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java [hunk 1]
```diff
@@ -58,11 +59,11 @@ private void sort() {
         while (true) {
             if (currentIndex < 0) break;
             IslandData islandData = getEntries().get(currentIndex);
-            if (islandData == null) {
+            if (island.equals(islandData.getIsland().orElse(null))) {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java:62]: islandData can be null, the check for that should stay.

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java [hunk 2]
```diff
@@ -26,18 +26,19 @@ public ForeignIslandTableManager(ConnectionSource connectionSource, Class<T> cla
 
     @Override
     public void addEntry(T t) {
-        getEntries().add(t);
+        int index = Collections.binarySearch(getEntries(), t, Comparator.comparing(t1 -> t1.getIsland().map(Island::getId).orElse(0)));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java:29]: This shouldnt be needed, it should add it in its correctly sorted place if you look in TableManager and SortedList class

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java [hunk 3]
```diff
@@ -26,18 +26,19 @@ public ForeignIslandTableManager(ConnectionSource connectionSource, Class<T> cla
...
+        getEntries().add(index < 0 ? -(index + 1) : index, t);
     }
 
     /**
      * Sort the list of entries by island id
      */
     private void sort() {
-        getEntries().sort(comparator);
+        getEntries().sort(Comparator.comparing(t1 -> t1.getIsland().map(Island::getId).orElse(0)));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/ForeignIslandTableManager.java:37]: This should be sorted using comparator? In Rewards case comparator is sorting by IslandID (which you can see in DatabaseManager)

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/IslandTableManager.java [hunk 4]
```diff
@@ -20,7 +20,8 @@ public IslandTableManager(ConnectionSource connectionSource) throws SQLException
 
     @Override
     public void addEntry(Island island) {
-        getEntries().add(island);
+        int index = Collections.binarySearch(getEntries(), island, Comparator.comparing(Island::getId));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/IslandTableManager.java:23]: Again this should be sorted using SortedList class

File: src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java [hunk 5]
```diff
@@ -38,8 +38,11 @@ public void sort() {
      * @param user The item we are adding
      */
     public void addEntry(User user) {
-        getEntries().add(user);
-        userIslandIndex.add(user);
+        int index = Collections.binarySearch(getEntries(), user, Comparator.comparing(User::getUuid));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java:41]: Again should be sorted using SortedList method

**Review by PeachesMLG — CHANGES_REQUESTED** [review]
Im kinda confused about what your trying to do here, this would break alot of changes and I dont see how it would fix missions

**Comment by PeachesMLG** [comment]
Replaced by: https://github.com/Iridium-Development/IridiumSkyblock/pull/412

**Pull request closed without merging**